test/unit/plugins/validate-semantic/oas3/refs.js

```diff
@@ -206,4 +206,84 @@ describe("validation plugin - semantic - oas3 refs", () => {
       return expectNoErrorsOrWarnings(spec)
     })
   })
+
+  describe("$refs for requestbody schemas must reference a schema by position", () => {
+    it("should return an error when a requestBody schema incorrectly references a local component schema", () => {
+      const spec = {
+        openapi: "3.0.0",
+        paths: {
+          "/foo": {
+            post: {
+              requestBody: {
+                content: {
+                  "application/json": {
+                    schema: {
+                      $ref: "#/components/requestbodies/Foo"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        },
+        components: {
+          schemas: {
+            Foo: {
+              type: "string"
+            }
+          }
+        }
+      }
+
+      return validateHelper(spec)
+        .then(system => {
+          const allErrors = system.errSelectors.allErrors().toJS()
+          expect(allErrors.length).toEqual(3)
```

Comment:
**Change request**: let's ignore the first two errors (which aren't semantic errors, and are unrelated to the code under test) by filtering them out.

You could do something like:

```js
const allErrors = system.errSelectors.allErrors().toJS()
const allSematicErrors = allErrors.filter(err => err.source === "semantic")

expect(allSemanticErrors.length).toEqual(1)

const firstError = allSemanticErrors[0]

expect(firstError.message).toEqual(`requestBody schemas $refs must point to a position where a schema can be legally placed`)
expect(firstError.path).toEqual(["paths", "/foo", "post", "requestBody", "content", "application/json", "schema", "$ref"])
```

Reply:
I tried with this filter, but still having 3 same errors. It seems all 3 are semantic errors.